File: Python3/1064.py
__________________________________________________________________________________________________
sample 40 ms submission
class Solution:
    def fixedPoint(self, A: List[int]) -> int:
        for i, n in enumerate(A):
            if i == n:
                return i
        return -1        
__________________________________________________________________________________________________
class Solution:
    def fixedPoint(self, A: List[int]) -> int:
        s,e = 0, len(A)-1
        while s < e:
            mid = (s+e)//2
            if A[mid]-mid<0: s += 1
            elif A[mid]-mid>0: e -= 1
            else: return mid
        return -1
__________________________________________________________________________________________________

File: C++/1064.cpp
__________________________________________________________________________________________________
sample 16 ms submission
class Solution {
public:
    int fixedPoint(vector<int>& A) {
        for(int i = 0;i < A.size();++i) {
            if(A[i] == i) {
                return i;
            }
        }
        return -1;
    }
};
__________________________________________________________________________________________________
class Solution {
public:
   int fixedPoint(vector<int>& A) {
        int l = 0, r = A.size() - 1;
        while (l <= r) {
            int m = (l + r) / 2;
            if (A[m] - m > 0)
                r = m - 1;
            else if (A[m] - m < 0)
                l = m + 1;
            else
                return m;
        }
        return -1;
    }
};
__________________________________________________________________________________________________
